implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -64,14 +66,30 @@ default String group() {
         return get(prefix() + ".group");
     }
 
+    /**
+     * Return the version of the target Dynatrace API.
+     *
+     * @return a {@link DynatraceApiVersion} containing the version of the targeted Dynatrace API.
+     */
+    default DynatraceApiVersion apiVersion() {
+        // if not specified or invalid, defaults to v1 for backwards compatibility.
+        return getEnum(this, DynatraceApiVersion.class, "apiVersion")
+                .orElse(DynatraceApiVersion.V1);
+    }
+
     @Override
     default Validated<?> validate() {
-        return checkAll(this,
-                c -> StepRegistryConfig.validate(c),
-                checkRequired("apiToken", DynatraceConfig::apiToken),
-                checkRequired("uri", DynatraceConfig::uri),
-                checkRequired("deviceId", DynatraceConfig::deviceId),
-                check("technologyType", DynatraceConfig::technologyType).andThen(Validated::nonBlank)
-        );
+        // Currently only v1 is implemented. This check will be extended once more versions are added.
+        if (apiVersion() == DynatraceApiVersion.V1) {
```

Comment:
This is much easier, and I adopted it for the v1 exporter. However, for the v2 exporter, we will only require the API version and checking these four required parameters with a `.andThen(v -> v.invalidateWhen(<check-version-here>)` construct for v1 seems a bit hard to read. Is there another way to make this more concise?

Reply:
InfluxConfix does something similar though, in this case, I would create a `validateV1` and a `validateV2` method (the current `validate` becomes `validateV1`). With them you can do:
```
@Override
default Validated<?> validate() {
    return apiVersion() == V1 ? validateV1() : validateV2();
}
```

Reply:
I split it out to a separate function, and also removed the invalidVersion test, as the test data is invalid and can never happen in a non-test environment. If the spring boot app is started with an application property of `v-INVALID`, for example, it will just crash on startup and provide all valid values (which is just `V1`) in this case.
Another way to get around this would be to explicitly validate the `apiVersion` before accessing it, something like the following: 
```
@Override
default Validated<?> validate() {
    Validated<DynatraceApiVersion> apiVersionValidation = checkRequired("apiVersion", DynatraceConfig::apiVersion).apply(this);
    if (apiVersionValidation.isInvalid()) {
        return apiVersionValidation;
    }
    return apiVersion() == V1 ? validateV1() : validateV2();
}
```
which seems a bit unnecessary if this case never actually appears during use.

Reply:
I just realized that that is only the case when using micrometer with Spring Boot. I think the solution I posted above (with the additional step of validating the apiVersion) should work, if i just access the `apiVersion()` method without validation first I recieve a `ValidationException`. I added the additional validation step.